pkgs/development/ocaml-modules/ppx_deriving/default.nix

```diff
@@ -1,20 +1,26 @@
 { lib, fetchzip, buildDunePackage
-, cppo, ppxfind, ppx_tools, ppx_derivers, result, ounit, ocaml-migrate-parsetree
+, cppo, ppx_derivers, ppxlib, ocaml-migrate-parsetree, result
+, ounit
 }:
 
 buildDunePackage rec {
   pname = "ppx_deriving";
-  version = "4.5";
+  version = "5.1";
+
+  useDune2 = true;
+
+  minimumOCamlVersion = "4.05";
```

Comment:
This restriction actually removes quite a few packages from nixpkgs. Is it a big deal? Adding also version 5.0 for OCaml ≤ 4.04 might not be too much of a burden.

Reply:
Done in 716d5f84d7861d21382a34946697861c9a786b9c, but haven't tested any 4.04 packages that depend on it yet.